lib/internal/crypto/hash.js

```diff
@@ -140,7 +145,53 @@ Hmac.prototype.digest = function digest(outputEncoding) {
 Hmac.prototype._flush = Hash.prototype._flush;
 Hmac.prototype._transform = Hash.prototype._transform;
 
+function Mac(mac, key, options) {
+  if (!(this instanceof Mac))
+    return new Mac(mac, key, options);
+
+  validateString(mac, 'mac');
+  key = prepareSecretKey(key);
+
+  let nid = EVP_PKEY_HMAC;
+  if (mac === 'poly1305') nid = EVP_PKEY_POLY1305;
+  else if (mac === 'siphash') nid = EVP_PKEY_SIPHASH;
+  else if (mac.startsWith('cmac:')) {
+    nid = EVP_PKEY_CMAC;
+    mac = mac.slice(5);
+  }
+
+  this[kHandle] = new _Mac(nid, toBuf(key), mac);
```

Comment:
```suggestion
  this[kHandle] = new _Mac(nid, key, mac);
```

Reply:
I left this out. Not being able to create a key from a string is just too annoying.